### PR TITLE
closePopup action yield

### PR DESCRIPTION
**Pull request by IAmJulianAcosta**

### CODE REVIEW

File: addon/mixins/popup.js [hunk 1]
```diff
@@ -9,6 +9,12 @@ export default Mixin.create({
 
   layout,
   popupOpen: false,
+  
+  actions: {
+    closePopup () {
```
Comment (miguelcobain) [inline addon/mixins/popup.js]: We don't need to make this an action. Let's make it a regular function.

File: addon/templates/current/popup.hbs [hunk 2]
```diff
@@ -1,4 +1,4 @@
 {{#if hasBlock}}{{setHasBlock}}{{/if}}
 {{#if popupOpen}}
-  {{yield}}
+  {{yield (action "closePopup")}}
```
Comment (miguelcobain) [inline addon/templates/current/popup.hbs]: If closePopup is a normal function here you can just `{{yield closePopup}}`.

**Comment by miguelcobain** [comment]
A test is missing.

**Comment by IAmJulianAcosta** [comment]
@miguelcobain if I pass a normal function, `this` context is lost. Any idea about how to fix it?

**Comment by miguelcobain** [comment]
@IAmJulianAcosta ah, really?

Maybe `{{yield (action closePopup)}}` (without quotes) probably works.

File: addon/mixins/popup.js [hunk 3]
```diff
@@ -10,6 +10,10 @@ export default Mixin.create({
   layout,
   popupOpen: false,
 
+  closePopup () {
+    this._popup._close ();
```
Comment (miguelcobain) [inline addon/mixins/popup.js]: there's a space after `close`

**Comment by IAmJulianAcosta** [comment]
What? And the test, I didn't write it! Will write it next week, deadline in 24 hours 😛

**Comment by miguelcobain** [comment]
@IAmJulianAcosta I wrote it. No worries. :)

**Pull request merged**